src/components/directory-content.js

```diff
@@ -563,10 +607,14 @@ const DirectoryContent = () => {
                             className={classes.circularRoot}
                         />
                     )}
-                {childrenMetadata[elementUuid] && getElementIcon(objectType)}
+                {childrenMetadata[elementUuid] &&
+                    getElementIcon(
+                        objectType,
+                        childrenMetadata[elementUuid].subtype
+                    )}
                 {/* Name */}
                 {isMetadataLoading ? null : childrenMetadata[elementUuid] ? (
-                    <div>{childrenMetadata[elementUuid].name}</div>
+                    <div>{elementName}</div>
```

Comment:
revert this, you should always prefer the name from the metadata I think

Reply:
done